packages/components/src/components/teaser/teaser.ts

```diff
@@ -0,0 +1,164 @@
+      this.resizeObserver.observe(this);
+    }
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    if (this.resizeObserver) {
+      this.resizeObserver.disconnect();
+    }
+  }
+
+  updated(changedProperties: PropertyValues) {
+    super.updated(changedProperties);
+    if (changedProperties.has('orientation')) {
+      this.updateOrientation();
+    }
+  }
+
+  updateOrientation() {
+    if (this.orientation === 'responsive') {
+      this.currentOrientation = this.offsetWidth > 448 ? 'horizontal' : 'vertical';
+    } else {
+      this.currentOrientation = this.orientation;
+    }
+  }
+
+  render() {
+    if (this.variant === 'white border-neutral-300') {
+      this.inset = true;
+    }
+
+    const slots = {
+      'teaser-has-default': this.hasSlotController.test('[default]'),
+      'teaser-has-media': this.hasSlotController.test('media'),
+      'teaser-has-meta': this.hasSlotController.test('meta'),
+      'teaser-has-headline': this.hasSlotController.test('headline')
+    };
+
+    return html`
+      <div
+        class=${cx(
+          'flex',
+          {
+            white: 'bg-white',
+            'white border-neutral-300': 'bg-white border border-neutral-300',
+            'neutral-100': 'bg-neutral-100',
+            primary: 'bg-primary text-white',
+            'primary-100': 'bg-primary-100'
+          }[this.variant],
+          this.currentOrientation === 'vertical' && 'flex-col',
+          this.currentOrientation === 'horizontal' && 'flex-row gap-8',
+          this.currentOrientation === 'horizontal' && this.inset && 'py-8 px-10'
+        )}
+        part="base"
+      >
+        <div
+          style=${this.currentOrientation === 'horizontal' ? `width: var(--distribution-media, 100%);` : ''}
+          class=${cx(
+            !this.inset && this.currentOrientation === 'vertical' && 'mb-4',
+            !slots['teaser-has-media'] && 'hidden'
+          )}
+          part="media"
+        >
+          <slot name="media"></slot>
+        </div>
+
+        <div
+          style=${this.currentOrientation === 'horizontal'
+            ? `width: var(--distribution-content, 100%); ${
+                this.inset ? 'width: var(--distribution-content, calc(100% - 2rem));' : ''
+              }`
+            : ''}
+          class=${cx(
+            'flex flex-col text-left',
+            this.currentOrientation === 'horizontal' && `flex flex-col`,
+            this.currentOrientation === 'vertical' && this.inset && 'm-4'
+          )}
+          id="content"
+        >
+          <div part="meta" class=${cx('gap-2 mb-4', !slots['teaser-has-meta'] && 'hidden')}>
+            <slot name="meta"></slot>
+          </div>
+
+          <div
+            part="headline"
+            class=${cx('text-lg font-bold mb-1', this.variant === 'primary' ? 'text-white' : 'text-black')}
+          >
+            <slot name="headline">Always insert one semantically correct heading element here &lt;h&gt;</slot>
```

Comment:
```suggestion
            <slot name="headline">Always insert one semantically correct heading element here (e. g. &lt;h2&gt;)</slot>
```

Sorry, I missed that. :D